rhel.Dockerfile

```diff
@@ -0,0 +1,129 @@
+#
+# Copyright (c) 2018-2019 Red Hat, Inc.
+# This program and the accompanying materials are made
+# available under the terms of the Eclipse Public License 2.0
+# which is available at https://www.eclipse.org/legal/epl-2.0/
+#
+# SPDX-License-Identifier: EPL-2.0
+#
+# Contributors:
+#   Red Hat, Inc. - initial API and implementation
+#
+
+# Builder: check meta.yamls and create index.json
+# https://access.redhat.com/containers/?tab=tags#/registry.access.redhat.com/ubi8-minimal
+FROM registry.access.redhat.com/ubi8-minimal:8.0-213 as builder
+USER 0
+
+################# 
+# PHASE ONE: create ubi8-minimal image with yq
+################# 
+
+# to get all the python deps pre-fetched so we can build in Brew:
+# 1. extract files in the container to your local filesystem
+#    CONTAINERNAME="pluginregistry"
+#    mkdir -p /tmp/root-local/ && docker run -it -v /tmp/root-local/:/tmp/root-local/ ${CONTAINERNAME} /bin/bash -c "cd /root/.local/ && cp -r bin/ lib/ /tmp/root-local/"
+# 2. then back on your own machine, tar them up and put them in this folder
+#    cd /tmp/root-local && sudo tar cvzf root-local.tgz lib/ bin/ && cd - && mv -f /tmp/root-local/root-local.tgz . && sudo rm -fr /tmp/root-local/
+# 3. then add it to dist-git so it's part of this repo
+#    rhpkg new-sources root-local.tgz 
+
+# built in Brew, use tarball in lookaside cache
+# COPY root-local.tgz /tmp/root-local.tgz
+
+# NOTE: uncomment for local build. Must also set full registry path in FROM to registry.redhat.io or registry.access.redhat.com
+# enable rhel 7 or 8 content sets (from Brew) to resolve jq as rpm
+COPY content_sets_epel7.repo /etc/yum.repos.d/
```

Comment:
WDYT about implementing this as an echo command? It's a weirder approach but also avoids adding the `content_sets_epel7.repo` file to the repo, which I think could be confusing for someone not familiar with rhel.

Alternatively, we could move dockerfiles and associated files into a `/build` directory in the repository, to make it clearer what belongs where.

Reply:
yes sounds weird to have these repo files in root directory

Reply:
+1 to use build/dockerfiles - consistency with https://github.com/eclipse/che-devfile-registry/tree/master/build/dockerfiles would be nice. I can move 'em if we agree.

As to the `echo > file` vs. `COPY` um... again, more consistent w/ up/downstream. In Brew this file isn't used (its info is injected by the Brew system). But to make these builds work locally outside Brew, the repo info is required. We could embed the information into a long echo line but OTOH if you concern is that people don't know stuff about rhel, then surely this would help them learn stuff, rather than hiding the details in the rhel.Dockerfile ?

Reply:
+1 on moving to `/build/dockerfiles`, and your explanation makes sense. Adding the content-sets file makes more sense to me if it's in a dockerfiles folder with the dockerfile.

Reply:
Here's the thing though... if I move the `*Dockerfile*` files to /build/dockerfiles, I also have to move ... `README.md`, `v3/` and `scripts/` since they have to be local to the Dockerfile context.

Which means the root folder then only has:

```
build/
build.sh
cico_build_ci.sh
cico_build_nightly.sh
cico_build_release.sh
cico_functions.sh
CONTRIBUTE.md
kubernetes/
LICENSE
openshift/
VERSION
```

I mean if you think having build-related files in the root is wrong, then maybe we need to move all the cico* files and build.sh too?

Reply:
Couldn't we specify the same context but put the dockerfile in a different dir?

```bash
docker build -t <image> -f ./build/dockerfiles/Dockerfile .
```

The context is still the root of the repo there.

Reply:
This is how we do it in the devfile registry: https://github.com/eclipse/che-devfile-registry/blob/master/build.sh#L20

Reply:
yeah, but can we handle that in a second PR? +1 as is if you like the current version and I'll propose a reorg change later so that I don't mess up the cico scripts too